spot-client/src/app.js

```diff
@@ -287,6 +290,23 @@ export class App extends React.Component {
         /** No-op. */
     }
 
+    /**
+     * Parses and preserves the query params the app was launched with for later use by
+     * their respective component.
+     *
+     * @returns {void}
+     */
+    _preserveQueryParams() {
+        const params = new URLSearchParams(this.props.location.search);
+        const rawParamsObject = {};
+
+        for (const [ key, value ] of params.entries()) {
+            rawParamsObject[key] = value;
+        }
+
+        this.props.dispatch(setStartParams(rawParamsObject));
```

Comment:
What do you think about shoving this stuff into the already cluttered bootstrap.js? Maybe where the default state is produced?

Reply:
Good idea, done! Although we may need to clean up that mess there in the future